src/plugins/intel_npu/src/common/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+# Copyright (C) 2018-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set(TARGET_NAME openvino_npu_common)
+
+file(GLOB_RECURSE SOURCES *.cpp *.hpp *.h)
+source_group(TREE ${CMAKE_CURRENT_SOURCE_DIR} FILES ${SOURCES})
+
+add_library(${TARGET_NAME} STATIC ${SOURCES})
+add_library(openvino::npu_common ALIAS ${TARGET_NAME})
+set_target_properties(${TARGET_NAME} PROPERTIES EXPORT_NAME npu_common)
+
+target_include_directories(${TARGET_NAME}
+    PUBLIC
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+)
+
+if(ENABLE_MLIR_COMPILER)
+    target_compile_definitions(${TARGET_NAME} PRIVATE ENABLE_MLIR_COMPILER)
+endif()
+
+target_link_libraries(${TARGET_NAME}
+    PUBLIC
+        openvino::npu_al
+        openvino::npu_logger_utils
+        openvino::runtime::dev
+)
+
+set_target_properties(${TARGET_NAME} PROPERTIES INTERPROCEDURAL_OPTIMIZATION_RELEASE ${ENABLE_LTO})
```

Comment:
Can't recall now, but there seem to be existing utility function/OV wrapper to enable LTO on given target?

Reply:
it is there for each target

Reply:
This is aligned with openvino core practice and with other plugins.